src/mem3/src/mem3_rep.erl

```diff
@@ -249,8 +248,9 @@ pull_purges(#acc{} = Acc0) ->
         end,
 
         if Remaining =< 0 -> ok; true ->
-            OldestPurgeSeq = couch_db:get_oldest_purge_seq(Db),
-            PurgesToPush = couch_db:get_purge_seq(Db) - OldestPurgeSeq,
+            {ok, PurgeSeq} = couch_db:get_purge_seq(Db),
```

Comment:
Did you verify it returns `{ok, pos_integer()}`? Even though dialyzer complains the tests are written in such a way that `ok` tag is not suppose to be there [see](https://github.com/apache/couchdb/blob/master/src/couch/src/test_engine_purge_docs.erl#L32). So either these tests are not running (which wouldn't surprise me given our previous experience with eunit) or the tag is added elsewere. Could you show me the lines where `{ok, integer()}` reply is formed? Also please verify that tests I pointed to are running.

Reply:
There are some differences between `couch_db:get_purge_seq/1` and `couch_bt_engine:get_purge_seq/1`. The first returns `{ok, PurgeSeq}` while the second returns `PurgeSeq`. The code for `couch_db:get_purge_seq/1` is 

```
get_purge_seq(#db{}=Db) ->
    {ok, couch_db_engine:get_purge_seq(Db)}.
```

Reply:
Thank you I found it [here](https://github.com/apache/couchdb/blob/COUCHDB-3326-clustered-purge-pr5-implementation/src/couch/src/couch_db.erl#L521)

This means that return values from couch_db are inconsistent. Here is a list of functions which deal with different sequences:

- `get_update_seq` returns `integer()`
- `get_purge_seq` returns `{ok, integer}`
- `get_compacted_seq` returns `integer()`
- `get_committed_update_seq` returns `integer()`

I think that since we are updating purge feature we should take an opportunity to make it consistent, change return type and update all callers.